modules/button/react/lib/ToolbarIconButton.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import {colors, spacing, borderRadius} from '@workday/canvas-kit-react-core';
+import {focusRing, useTheme, Themeable, EmotionCanvasTheme} from '@workday/canvas-kit-react-common';
+import {SystemIcon} from '@workday/canvas-kit-react-icon';
+import {CanvasSystemIcon} from '@workday/design-assets-types';
+import {ButtonColors} from './types';
+import {ButtonContainer} from './parts';
+import {IconButtonProps} from './IconButton';
+
+export interface ToolbarIconButtonProps
+  extends Omit<IconButtonProps, 'size' | 'variant'>,
+    Themeable {
+  /**
+   * The accessibility label to indicate the action triggered by clicking the IconButton.
+   */
+  'aria-label': string;
+  /**
+   * The toggled state of the button. If defined as a boolean, then it manages the toggled state: on (`true`) or off (`false`).
+   * @default undefined
+   */
+  toggled?: boolean;
+  /**
+   * The ref to the button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * The icon of the IconButton.
+   */
+  icon?: CanvasSystemIcon;
+  /**
+   * The function called when the IconButton toggled state changes.
+   */
+  onToggleChange?: (toggled: boolean | undefined) => void;
```

Comment:
I removed the `as` prop. Otherwise I had to use `ButtonOrAnchor` and that requires  a variant and size, but that's not relevant in this specific component, thoughts?